test/t8_forest/t8_gtest_forest_new_from_part_cmesh.cxx

```diff
@@ -0,0 +1,234 @@
+  // an equidistributed uniform mesh
+  //   t8_forest_t forest = t8_forest_new_uniform (cmesh, t8_scheme_new_default_cxx (), 0, 1, sc_MPI_COMM_WORLD);
+  t8_forest_unref (&forest);
+}
+
+TEST (t8_forest_new_from_partition, two_one_quad)
+{
+  /* Current process */
+  int rank, size;
+  sc_MPI_Comm_size (sc_MPI_COMM_WORLD, &size);
+  sc_MPI_Comm_rank (sc_MPI_COMM_WORLD, &rank);
+
+  /* Geometry */
+  int degree = 2;
+  std::vector<std::vector<double>> cells;
+  cells.push_back (std::vector<double> { -1, 1.2, 0,   0, 1.2,  0,   -1, 3,    0, 0, 3,    0,   -1, 2.1,
+                                         0,  0,   2.1, 0, -0.5, 1.2, 0,  -0.5, 3, 0, -0.5, 2.1, 0 });
+  cells.push_back (std::vector<double> { 0, 1.2, 0,   2, 1.5, 0,    0, 3, 0, 2, 3,   0,   0, 2.1,
+                                         0, 1.7, 2.1, 0, 1,   1.55, 0, 1, 3, 0, 0.9, 2.1, 0 });
+  cells.push_back (
+    std::vector<double> { 2, 1.5, 0, 4, 0, 0, 2, 3, 0, 4, 3, 0, 1.7, 2.1, 0, 4, 1.5, 0, 3, 1, 0, 3, 3, 0, 3, 2, 0 });
+
+  /* Create a partitioned cmesh */
+  t8_cmesh_t cmesh;
+  t8_cmesh_init (&cmesh);
+  if (size != 2)
+    SC_ABORT ("Program must be run on two MPI processes.\n");
+  else {
+    if (rank == 0) {
+      for (size_t itree = 0; itree < 3; ++itree) {
+        t8_cmesh_set_tree_class (cmesh, itree, T8_ECLASS_QUAD);
+        t8_cmesh_set_tree_vertices (cmesh, itree, cells[itree].data (), cells[itree].size ());
+        t8_cmesh_set_attribute (cmesh, itree, t8_get_package_id (), T8_CMESH_LAGRANGE_POLY_DEGREE, &degree,
+                                sizeof (int), 1);
+      }
+      t8_cmesh_set_join (cmesh, 0, 1, 1, 0, 0);
+      t8_cmesh_set_join (cmesh, 1, 2, 1, 0, 0);
+      t8_cmesh_set_partition_range (cmesh, 3, 0, 1);
+    }
+    else if (rank == 1) {
+      t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_QUAD);
+      t8_cmesh_set_tree_class (cmesh, 2, T8_ECLASS_QUAD);
+      t8_cmesh_set_tree_vertices (cmesh, 1, cells[1].data (), cells[1].size ());
+      t8_cmesh_set_tree_vertices (cmesh, 2, cells[2].data (), cells[2].size ());
+      t8_cmesh_set_attribute (cmesh, 1, t8_get_package_id (), T8_CMESH_LAGRANGE_POLY_DEGREE, &degree, sizeof (int), 1);
+      t8_cmesh_set_attribute (cmesh, 2, t8_get_package_id (), T8_CMESH_LAGRANGE_POLY_DEGREE, &degree, sizeof (int), 1);
+      t8_cmesh_set_join (cmesh, 1, 2, 1, 0, 0);
+      t8_cmesh_set_partition_range (cmesh, 3, 2, 2);
+    }
+  }
+  t8_cmesh_register_geometry<t8_geometry_lagrange> (cmesh, 2);
+  t8_cmesh_commit (cmesh, sc_MPI_COMM_WORLD);
+  t8_forest_t forest
+    = t8_forest_new_uniform_with_cmesh_partition (cmesh, t8_scheme_new_default_cxx (), 0, 0, sc_MPI_COMM_WORLD);
+  // The following line fails, as the refinement cmesh partitioned required from t8code does not match the provided partition.
+  //  t8_forest_t forest = t8_forest_new_uniform (cmesh, t8_scheme_new_default_cxx (), 0, 1, sc_MPI_COMM_WORLD);
+  t8_forest_unref (&forest);
+}
```

Comment:
The test look very similar to each other. Is there a good way to avoid code-duplication?